chapter04/jxp/ch4ex4.pl

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use v5.10;
+
+sub greet {
+  state $previous;
+  my $name = shift;
+  if( !defined $previous ){
+    say "Hi $name! You are the first one here!";
+    $previous = $name;
+  } else {
+    say "Hi $name! $previous is also here!";
+    $previous = $name;
+  }
```

Comment:
<code> $previous = $name </code> can be outside the if/else loop. Saves duplication of code.